.github/workflows/sycl-detect-changes.yml

```diff
@@ -69,6 +86,15 @@ jobs:
               return '${{ steps.changes.outputs.changes }}';
             }
             // Treat everything as changed for huge PRs.
-            return ["llvm", "llvm_spirv", "clang", "sycl_fusion", "xptifw", "libclc", "sycl", "ci"];
+            return ["llvm", "llvm_spirv", "clang", "sycl_fusion", "xptifw", "libclc", "sycl", "ci", "esimd"];
 
       - run: echo '${{ steps.result.outputs.result }}'
+
+      - name: Set Arc tests
+        id: arc_tests
+        run: |
+          if [ "${{ contains(steps.result.outputs.result, 'esimd') }}" == "true" ]; then
+            echo 'arc_tests="(ESIMD|InvokeSimd|Matrix)/"' >> "$GITHUB_OUTPUT"
+          else
+            echo 'arc_tests="Matrix/"' >> "$GITHUB_OUTPUT"
+          fi
```

Comment:
Hm, I've just realised that it's not in the precommit yml file. What was the reason? We haven't had any real logic here before this change.

Reply:
I felt it's related more to processing the changes list which we already do in that yml, and the complexity of the logic seems to be similar to the existing 'Set output' job in the same file. I can move it if you'd prefer but I'd ask to like to do it in a separate PR because my team needs this testing ASAP.

Reply:
Yes, I'd like it moved. The reason is that I want this file to detect undisputable facts only and the tests we want to run on a given runner is "subjective" in nature. I'm ok with doing it in the next PR.

Reply:
Thanks I will do it immediate after this is merged